src/plugins/intel_gpu/src/kernel_selector/tensor_type.cpp

```diff
@@ -534,11 +534,22 @@ void DataTensor::SwapXY() {
     // Swap XY axes.
     y.pitch = 1;
     x.pitch = y.v + y.pad.Total();
+    int x_idx = Channelndex(l, DataChannelName::X);
+    int y_idx = Channelndex(l, DataChannelName::Y);
+    int f_idx = Channelndex(l, DataChannelName::FEATURE);
+    int b_idx = Channelndex(l, DataChannelName::BATCH);
+
+    if (x_idx < 0 || y_idx < 0 || f_idx < 0 || b_idx < 0) {
+        throw std::runtime_error("Invalid layout channel index");
+    }
+
     std::vector<Dim> vec(ChannelsCount(l));
-    vec[Channelndex(l, DataChannelName::X)] = y;
-    vec[Channelndex(l, DataChannelName::Y)] = x;
-    vec[Channelndex(l, DataChannelName::FEATURE)] = Feature();
-    vec[Channelndex(l, DataChannelName::BATCH)] = Batch();
+
+    vec[static_cast<size_t>(x_idx)] = y;
+    vec[static_cast<size_t>(y_idx)] = x;
+    vec[static_cast<size_t>(f_idx)] = Feature();
+    vec[static_cast<size_t>(b_idx)] = Batch();
+
```

Comment:
is it necessary to typecast into `size_t` to use as vector index?

Reply:
No longer need to check them because they have been checked right above.